sdks/java/core/src/test/java/org/apache/beam/sdk/testing/TestPipelineTest.java

```diff
@@ -139,6 +141,15 @@ public void testMatcherSerializationDeserialization() {
     assertEquals(m2, newOpts.getOnSuccessMatcher());
   }
 
+  @Test
```

Comment:
Part of the point, though, is that it does not fail if run() is not called, so I'd explicitly test that.